Running tests under nox is flaxy

I'm trying to run the tests. I created a virtualenv for installing nox and installed the prescribed version, but the tests fail:

```
pipx master $ python -m venv .nox/.nox                                                                                                                                    
pipx master $ .nox/.nox/bin/pip install --use-pep517 setuptools 'git+https://github.com/cs01/nox.git@5ea70723e9e6#egg=nox'                                                
Requirement already satisfied: setuptools in ./.nox/.nox/lib/python3.8/site-packages (41.2.0)
Collecting nox from git+https://github.com/cs01/nox.git@5ea70723e9e6#egg=nox
  Cloning https://github.com/cs01/nox.git (to revision 5ea70723e9e6) to /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-cr4qr61n/nox
  Running command git clone -q https://github.com/cs01/nox.git /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-cr4qr61n/nox
  WARNING: Did not find branch or tag '5ea70723e9e6', assuming revision or ref.
  Running command git checkout -q 5ea70723e9e6
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting virtualenv>=14.0.0 (from nox)
  Using cached https://files.pythonhosted.org/packages/05/f1/2e07e8ca50e047b9cc9ad56cf4291f4e041fa73207d000a095fe478abf84/virtualenv-16.7.9-py2.py3-none-any.whl
Collecting py<2.0.0,>=1.4.0 (from nox)
  Using cached https://files.pythonhosted.org/packages/99/8d/21e1767c009211a62a8e3067280bfce76e89c9f876180308515942304d2d/py-1.8.1-py2.py3-none-any.whl
Collecting colorlog<4.0.0,>=2.6.1 (from nox)
  Using cached https://files.pythonhosted.org/packages/d7/00/defed2a2eb8e289a0b6f29b3bcc52b42a75ce7cb14a84687c4c54d58f08e/colorlog-3.2.0-py2.py3-none-any.whl
Collecting argcomplete<2.0,>=1.9.4 (from nox)
  Using cached https://files.pythonhosted.org/packages/82/7d/455e149c28c320044cb763c23af375bd77d52baca041f611f5c2b4865cf4/argcomplete-1.11.1-py2.py3-none-any.whl
Building wheels for collected packages: nox
  Building wheel for nox (PEP 517) ... done
  Created wheel for nox: filename=nox-2019.5.30-cp38-none-any.whl size=37908 sha256=7518b21327b772a2e9897ab52147d455d7c239bf138e56817c7da43cfe444e0a
  Stored in directory: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-ephem-wheel-cache-50xn9u3j/wheels/1d/aa/ed/a061887cc0a8c9bedfe294c95ff7f75ff6771ec85834ec35cc
Successfully built nox
Installing collected packages: virtualenv, py, colorlog, argcomplete, nox
Successfully installed argcomplete-1.11.1 colorlog-3.2.0 nox-2019.5.30 py-1.8.1 virtualenv-16.7.9
WARNING: You are using pip version 19.2.3, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
pipx master $ .nox/.nox/bin/python -m nox --session tests-3.8                                                                                                             
nox > Running session tests-3.8
nox > Creating venv using python3.8 in .nox/tests-3-8
nox > pip install -e . pytest pytest-cov
nox > pytest --cov=pipx --cov-config .coveragerc --cov-report= tests
========================================================================== test session starts ===========================================================================
platform darwin -- Python 3.8.1, pytest-5.3.4, py-1.8.1, pluggy-0.13.1
rootdir: /Users/jaraco/code/public/pipx
plugins: cov-2.8.1
collected 75 items                                                                                                                                                       

tests/test_animate.py ...                                                                                                                                          [  4%]
tests/test_completions.py .                                                                                                                                        [  5%]
tests/test_inject.py F.F                                                                                                                                           [  9%]
tests/test_install.py ..F...............                                                                                                                           [ 33%]
tests/test_list.py ..                                                                                                                                              [ 36%]
tests/test_main.py .......                                                                                                                                         [ 45%]
tests/test_pipx_metadata_file.py .....F                                                                                                                            [ 53%]
tests/test_reinstall_all.py .                                                                                                                                      [ 54%]
tests/test_run.py ...F......................                                                                                                                       [ 89%]
tests/test_runpip.py .                                                                                                                                             [ 90%]
tests/test_shared_libs.py ..                                                                                                                                       [ 93%]
tests/test_uninstall.py ..                                                                                                                                         [ 96%]
tests/test_uninstall_all.py .                                                                                                                                      [ 97%]
tests/test_upgrade.py .                                                                                                                                            [ 98%]
tests/test_upgrade_all.py .                                                                                                                                        [100%]Coverage.py warning: Data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.49499.016930' doesn't seem to be a coverage data file: Couldn't use data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.49499.016930': no such table: coverage_schema
Coverage.py warning: Data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.48749.385928' doesn't seem to be a coverage data file: Couldn't use data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.48749.385928': no such table: coverage_schema
Coverage.py warning: Data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.49498.790109' doesn't seem to be a coverage data file: Couldn't use data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.49498.790109': no such table: coverage_schema
Coverage.py warning: Data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.49496.557321' doesn't seem to be a coverage data file: Couldn't use data file '/Users/jaraco/code/public/pipx/.coverage.Karui-Tatchi.lan.49496.557321': no such table: coverage_schema


================================================================================ FAILURES ================================================================================
______________________________________________________________________________ test_simple _______________________________________________________________________________

pipx_temp_env = None, capsys = <_pytest.capture.CaptureFixture object at 0x108060700>

    def test_simple(pipx_temp_env, capsys):
        assert not run_pipx_cli(["install", "pycowsay"])
>       assert not run_pipx_cli(["inject", "pycowsay", "black"])
E       AssertionError: assert not 1
E        +  where 1 = run_pipx_cli(['inject', 'pycowsay', 'black'])

tests/test_inject.py:7: AssertionError
-------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------
  installed package pycowsay 0.0.0.1, Python 3.8.1
  These apps are now globally available
    - pycowsay
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
  ERROR: Command errored out with exit status 1:
   command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-wheel-jsaoxduv
       cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/
  Complete output (21 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  running build_ext
  building 'regex._regex' extension
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/regex_3
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
  In file included from regex_3/_regex.c:48:
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
  #include <stdio.h>
           ^~~~~~~~~
  1 error generated.
  error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for regex
    ERROR: Command errored out with exit status 1:
     command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-g90akhho/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/include/site/python3.8/regex
         cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/
    Complete output (21 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.8
    creating build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    running build_ext
    building 'regex._regex' extension
    creating build/temp.macosx-10.9-x86_64-3.8
    creating build/temp.macosx-10.9-x86_64-3.8/regex_3
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
    In file included from regex_3/_regex.c:48:
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
    #include <stdio.h>
             ^~~~~~~~~
    1 error generated.
    error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-9jo471ao/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-g90akhho/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_simple0/subdir/pipxhome/venvs/pycowsay/include/site/python3.8/regex Check the logs for full command output.
done! ✨ 🌟 ✨
Error installing black.
___________________________________________________________________________ test_include_apps ____________________________________________________________________________

pipx_temp_env = None, capsys = <_pytest.capture.CaptureFixture object at 0x10808ac10>

    def test_include_apps(pipx_temp_env, capsys):
        assert not run_pipx_cli(["install", "pycowsay"])
        assert run_pipx_cli(["inject", "pycowsay", "black", "--include-deps"])
>       assert not run_pipx_cli(
            ["inject", "pycowsay", "black", "--include-deps", "--include-apps"]
        )
E       AssertionError: assert not 1
E        +  where 1 = run_pipx_cli(['inject', 'pycowsay', 'black', '--include-deps', '--include-apps'])

tests/test_inject.py:18: AssertionError
-------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------
  installed package pycowsay 0.0.0.1, Python 3.8.1
  These apps are now globally available
    - pycowsay
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
  ERROR: Command errored out with exit status 1:
   command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-wheel-923xaxm4
       cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/
  Complete output (21 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  running build_ext
  building 'regex._regex' extension
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/regex_3
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
  In file included from regex_3/_regex.c:48:
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
  #include <stdio.h>
           ^~~~~~~~~
  1 error generated.
  error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for regex
    ERROR: Command errored out with exit status 1:
     command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-9k4opfzx/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/include/site/python3.8/regex
         cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/
    Complete output (21 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.8
    creating build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    running build_ext
    building 'regex._regex' extension
    creating build/temp.macosx-10.9-x86_64-3.8
    creating build/temp.macosx-10.9-x86_64-3.8/regex_3
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
    In file included from regex_3/_regex.c:48:
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
    #include <stdio.h>
             ^~~~~~~~~
    1 error generated.
    error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-izbn30m3/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-9k4opfzx/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_include_apps0/subdir/pipxhome/venvs/pycowsay/include/site/python3.8/regex Check the logs for full command output.
done! ✨ 🌟 ✨
Cannot pass --include-deps if --include-apps is not passed as well
Error installing black.
___________________________________________________________________ test_install_easy_packages[black] ____________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x1082551c0>, pipx_temp_env = None, caplog = <_pytest.logging.LogCaptureFixture object at 0x1082758b0>
package = 'black'

    @pytest.mark.parametrize("package", ["pycowsay", "black"])
    def test_install_easy_packages(capsys, pipx_temp_env, caplog, package):
>       install_package(capsys, pipx_temp_env, caplog, package)

tests/test_install.py:48: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

capsys = <_pytest.capture.CaptureFixture object at 0x1082551c0>, pipx_temp_env = None, caplog = <_pytest.logging.LogCaptureFixture object at 0x1082758b0>
package = 'black', package_name = 'black'

    def install_package(capsys, pipx_temp_env, caplog, package, package_name=""):
        if not package_name:
            package_name = package
    
        run_pipx_cli(["install", package, "--verbose"])
        captured = capsys.readouterr()
>       assert f"installed package {package_name}" in captured.out
E       AssertionError: assert 'installed package black' in '\n'
E        +  where '\n' = CaptureResult(out='\n', err='Error installing black.\n').out

tests/test_install.py:34: AssertionError
-------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------
Collecting black
  Using cached black-19.10b0-py36-none-any.whl (97 kB)
Collecting appdirs
  Using cached appdirs-1.4.3-py2.py3-none-any.whl (12 kB)
Collecting pathspec<1,>=0.6
  Using cached pathspec-0.7.0-py2.py3-none-any.whl (25 kB)
Collecting click>=6.5
  Using cached Click-7.0-py2.py3-none-any.whl (81 kB)
Collecting attrs>=18.1.0
  Using cached attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Collecting toml>=0.9.4
  Using cached toml-0.10.0-py2.py3-none-any.whl (25 kB)
Collecting regex
  Using cached regex-2020.1.8.tar.gz (681 kB)
Collecting typed-ast>=1.4.0
  Using cached typed_ast-1.4.1-cp38-cp38-macosx_10_15_x86_64.whl (224 kB)
Building wheels for collected packages: regex
  Building wheel for regex (setup.py): started
  Building wheel for regex (setup.py): finished with status 'error'
  Running setup.py clean for regex
Failed to build regex
Installing collected packages: appdirs, pathspec, click, attrs, toml, regex, typed-ast, black
    Running setup.py install for regex: started
    Running setup.py install for regex: finished with status 'error'
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
  ERROR: Command errored out with exit status 1:
   command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-wheel-w1byycdn
       cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/
  Complete output (21 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  running build_ext
  building 'regex._regex' extension
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/regex_3
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
  In file included from regex_3/_regex.c:48:
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
  #include <stdio.h>
           ^~~~~~~~~
  1 error generated.
  error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for regex
    ERROR: Command errored out with exit status 1:
     command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-uu1ef87n/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/include/site/python3.8/regex
         cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/
    Complete output (21 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.8
    creating build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    running build_ext
    building 'regex._regex' extension
    creating build/temp.macosx-10.9-x86_64-3.8
    creating build/temp.macosx-10.9-x86_64-3.8/regex_3
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
    In file included from regex_3/_regex.c:48:
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
    #include <stdio.h>
             ^~~~~~~~~
    1 error generated.
    error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-2yspkg39/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-uu1ef87n/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_install_easy_packages_bla0/subdir/pipxhome/venvs/black/include/site/python3.8/regex Check the logs for full command output.
__________________________________________________________________________ test_package_inject ___________________________________________________________________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x108342fd0>
tmp_path = PosixPath('/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0'), pipx_temp_env = None

    def test_package_inject(monkeypatch, tmp_path, pipx_temp_env):
        pipx_venvs_dir = pipx.constants.PIPX_HOME / "venvs"
    
        run_pipx_cli(["install", "pycowsay"])
        run_pipx_cli(["inject", "pycowsay", "black"])
        assert (pipx_venvs_dir / "pycowsay" / "pipx_metadata.json").is_file()
    
        pipx_metadata = PipxMetadata(pipx_venvs_dir / "pycowsay")
    
>       assert pipx_metadata.injected_packages.keys() == {"black"}
E       AssertionError: assert dict_keys([]) == {'black'}
E         Use -v to get the full diff

tests/test_pipx_metadata_file.py:157: AssertionError
-------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------
  installed package pycowsay 0.0.0.1, Python 3.8.1
  These apps are now globally available
    - pycowsay
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
done! ✨ 🌟 ✨
  ERROR: Command errored out with exit status 1:
   command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-wheel-y9nllr3g
       cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/
  Complete output (21 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  running build_ext
  building 'regex._regex' extension
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/regex_3
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
  In file included from regex_3/_regex.c:48:
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
  #include <stdio.h>
           ^~~~~~~~~
  1 error generated.
  error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for regex
    ERROR: Command errored out with exit status 1:
     command: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-dxqtxbe3/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/include/site/python3.8/regex
         cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/
    Complete output (21 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.8
    creating build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
    running build_ext
    building 'regex._regex' extension
    creating build/temp.macosx-10.9-x86_64-3.8
    creating build/temp.macosx-10.9-x86_64-3.8/regex_3
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
    In file included from regex_3/_regex.c:48:
    /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
    #include <stdio.h>
             ^~~~~~~~~
    1 error generated.
    error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/setup.py'"'"'; __file__='"'"'/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-alpx7q3b/regex/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-record-dxqtxbe3/install-record.txt --single-version-externally-managed --compile --install-headers /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-17/test_package_inject0/subdir/pipxhome/venvs/pycowsay/include/site/python3.8/regex Check the logs for full command output.
Error installing black.
_____________________________________________________________________ test_run_script_from_internet ______________________________________________________________________

pipx_temp_env = None, capsys = <_pytest.capture.CaptureFixture object at 0x1082b1040>

    def test_run_script_from_internet(pipx_temp_env, capsys):
>       assert not run_pipx_cli(
            [
                "run",
                "https://gist.githubusercontent.com/cs01/"
                "fa721a17a326e551ede048c5088f9e0f/raw/"
                "6bdfbb6e9c1132b1c38fdd2f195d4a24c540c324/pipx-demo.py",
            ]
        )
E       AssertionError: assert not 1
E        +  where 1 = run_pipx_cli(['run', 'https://gist.githubusercontent.com/cs01/fa721a17a326e551ede048c5088f9e0f/raw/6bdfbb6e9c1132b1c38fdd2f195d4a24c540c324/pipx-demo.py'])

tests/test_run.py:42: AssertionError
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
<urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1108)>


================================================================ 5 failed, 70 passed in 202.51s (0:03:22) ================================================================
nox > Command pytest --cov=pipx --cov-config .coveragerc --cov-report= tests failed with exit code 1
nox > Session tests-3.8 failed.
```

Since the issue doesn't appear to be with how nox configures its environment, I started looking into the test suite, and I found [the pipx_temp_env fixture](https://github.com/pipxproject/pipx/blob/2bb384acb12921dbdb3512e41fec58f2e9503959/tests/conftest.py#L9-L25), which is explicitly monkey-patching the environment in which packages are installed. I guess it's no big surprise if the test suite is removing all binaries from PATH that a compiler will fail to execute.

Indeed, I'm able to replicate the failure mode by simply spamming path before attempting to install regex:

```
pipx master $ env PATH='/does-not-exist' /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 -m pip install --use-pep517 --no-cache-dir regex                   
Collecting regex
  Downloading regex-2020.1.8.tar.gz (681 kB)
     |████████████████████████████████| 681 kB 1.4 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Building wheels for collected packages: regex
  Building wheel for regex (PEP 517) ... error
Failed to build regex
  ERROR: Command errored out with exit status 1:
   command: /Library/Frameworks/Python.framework/Versions/3.8/bin/python3 /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/tmpq7xizudo
       cwd: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-install-eu_udmuq/regex
  Complete output (21 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/_regex_core.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  copying regex_3/test_regex.py -> build/lib.macosx-10.9-x86_64-3.8/regex
  running build_ext
  building 'regex._regex' extension
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/regex_3
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c regex_3/_regex.c -o build/temp.macosx-10.9-x86_64-3.8/regex_3/_regex.o
  In file included from regex_3/_regex.c:48:
  /Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/Python.h:25:10: fatal error: 'stdio.h' file not found
  #include <stdio.h>
           ^~~~~~~~~
  1 error generated.
  error: command '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for regex
ERROR: Could not build wheels for regex which use PEP 517 and cannot be installed directly
```

My guess is there's something about the compiler that requires other tools on the path to be present for the include directories to be resolved correctly.

Is it reasonable for the pipx tests to clobber the path?

In 5fa0243, I found that by not clobbering the path, the tests stop failing... but some new tests start failing because I have black installed elsewhere, so triggers warnings (and assertions) about that:

```
___________________________________________ test_install_package_specs[black-https://github.com/ambv/black/archive/18.9b0.zip] ___________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x10c5ab670>, pipx_temp_env = None, caplog = <_pytest.logging.LogCaptureFixture object at 0x10c60ea90>
package_name = 'black', package_spec = 'https://github.com/ambv/black/archive/18.9b0.zip'

    @pytest.mark.parametrize(
        "package_name,package_spec",
        [
            # ("nox", "git+https://github.com/cs01/nox.git@5ea70723e9e6"),
            ("pylint", "pylint==2.3.1"),
            ("black", "https://github.com/ambv/black/archive/18.9b0.zip"),
        ],
    )
    def test_install_package_specs(
        capsys, pipx_temp_env, caplog, package_name, package_spec
    ):
>       install_package(capsys, pipx_temp_env, caplog, package_spec, package_name)

tests/test_install.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

capsys = <_pytest.capture.CaptureFixture object at 0x10c5ab670>, pipx_temp_env = None, caplog = <_pytest.logging.LogCaptureFixture object at 0x10c60ea90>
package = 'https://github.com/ambv/black/archive/18.9b0.zip', package_name = 'black'

    def install_package(capsys, pipx_temp_env, caplog, package, package_name=""):
        if not package_name:
            package_name = package
    
        run_pipx_cli(["install", package, "--verbose"])
        captured = capsys.readouterr()
        assert f"installed package {package_name}" in captured.out
        if not sys.platform.startswith("win"):
            # TODO assert on windows too
            # https://github.com/pipxproject/pipx/issues/217
            assert "symlink missing or pointing to unexpected location" not in captured.out
        assert "not modifying" not in captured.out
        assert "is not on your PATH environment variable" not in captured.out
        for record in caplog.records:
>           assert "⚠️" not in record.message
E           AssertionError: assert '⚠️' not in '⚠️  Note: b...al/bin/black'
E             '⚠️' is contained here:
E               ⚠️  Note: black was already on your PATH at /Users/jaraco/.local/bin/black
E             ? ++

tests/test_install.py:42: AssertionError
-------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------
Collecting https://github.com/ambv/black/archive/18.9b0.zip
  Using cached https://github.com/ambv/black/archive/18.9b0.zip
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Building wheels for collected packages: black
  Building wheel for black (PEP 517): started
  Building wheel for black (PEP 517): finished with status 'done'
  Created wheel for black: filename=black-18.9b0-py36-none-any.whl size=88091 sha256=08f98cb8798aa832e85f128e52f4f43d6e3134a955dd03a607f87a88ca674db1
  Stored in directory: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-ephem-wheel-cache-6xbmwu0o/wheels/c2/6a/a3/93788da15f8a518472c6a36fa9fd1d5ae3f686c81c0a4d90b1
Successfully built black
Installing collected packages: black
Successfully installed black-18.9b0
Collecting https://github.com/ambv/black/archive/18.9b0.zip
  Using cached https://github.com/ambv/black/archive/18.9b0.zip
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Collecting attrs>=17.4.0
  Using cached attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Collecting appdirs
  Using cached appdirs-1.4.3-py2.py3-none-any.whl (12 kB)
Collecting click>=6.5
  Using cached Click-7.0-py2.py3-none-any.whl (81 kB)
Collecting toml>=0.9.4
  Using cached toml-0.10.0-py2.py3-none-any.whl (25 kB)
Building wheels for collected packages: black
  Building wheel for black (PEP 517): started
  Building wheel for black (PEP 517): finished with status 'done'
  Created wheel for black: filename=black-18.9b0-py36-none-any.whl size=88091 sha256=bf8ebce5889820f4a0673add55736793e5817378fec2d7bcbc6ab57719176ac7
  Stored in directory: /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-ephem-wheel-cache-qdgjatb0/wheels/c2/6a/a3/93788da15f8a518472c6a36fa9fd1d5ae3f686c81c0a4d90b1
Successfully built black
Installing collected packages: attrs, appdirs, click, toml, black
Successfully installed appdirs-1.4.3 attrs-19.3.0 black-18.9b0 click-7.0 toml-0.10.0
--------------------------------------------------------------------------- Captured log call ----------------------------------------------------------------------------
WARNING  root:common.py:79 ⚠️  Note: black was already on your PATH at /Users/jaraco/.local/bin/black
WARNING  root:common.py:79 ⚠️  Note: blackd was already on your PATH at /Users/jaraco/.local/bin/blackd
```

And an error about certificate verification failing (probably another clobbered environment setting for those certificates):

```
_____________________________________________________________________ test_run_script_from_internet ______________________________________________________________________

pipx_temp_env = None, capsys = <_pytest.capture.CaptureFixture object at 0x10c5e2ee0>

    def test_run_script_from_internet(pipx_temp_env, capsys):
>       assert not run_pipx_cli(
            [
                "run",
                "https://gist.githubusercontent.com/cs01/"
                "fa721a17a326e551ede048c5088f9e0f/raw/"
                "6bdfbb6e9c1132b1c38fdd2f195d4a24c540c324/pipx-demo.py",
            ]
        )
E       AssertionError: assert not 1
E        +  where 1 = run_pipx_cli(['run', 'https://gist.githubusercontent.com/cs01/fa721a17a326e551ede048c5088f9e0f/raw/6bdfbb6e9c1132b1c38fdd2f195d4a24c540c324/pipx-demo.py'])

tests/test_run.py:42: AssertionError
-------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
<urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1108)>

```

--- a/.github/workflows/on-push.yml
+++ b/.github/workflows/on-push.yml
@@ -70,11 +70,11 @@ jobs:
       id: pip-cache
       run: |
         echo "::set-output name=dir::$(pip cache dir)"
-    - name: Persistent Github pip cache
-      uses: actions/cache@v2
-      with:
-        path: ${{ steps.pip-cache.outputs.dir }}
-        key: ${{ runner.os }}-pip${{ matrix.python-version }}
+        # - name: Persistent Github pip cache
+        #   uses: actions/cache@v2
+        #   with:
+        #     path: ${{ steps.pip-cache.outputs.dir }}
+        #     key: ${{ runner.os }}-pip${{ matrix.python-version }}
     - name: Upgrade pip
       run: python -m pip install --upgrade pip
     - name: Install nox

--- a/.github/workflows/on-push.yml
+++ b/.github/workflows/on-push.yml
@@ -70,11 +70,11 @@ jobs:
       id: pip-cache
       run: |
         echo "::set-output name=dir::$(pip cache dir)"
-        #- name: Persistent Github pip cache
-        #  uses: actions/cache@v2
-        #  with:
-        #    path: ${{ steps.pip-cache.outputs.dir }}
-        #    key: ${{ runner.os }}-pip${{ matrix.python-version }}
+    - name: Persistent Github pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: ${{ runner.os }}-pip${{ matrix.python-version }}
     - name: Upgrade pip
       run: python -m pip install --upgrade pip
     - name: Install nox

--- a/.github/workflows/on-push.yml
+++ b/.github/workflows/on-push.yml
@@ -70,11 +70,11 @@ jobs:
       id: pip-cache
       run: |
         echo "::set-output name=dir::$(pip cache dir)"
-    - name: Persistent Github pip cache
-      uses: actions/cache@v2
-      with:
-        path: ${{ steps.pip-cache.outputs.dir }}
-        key: ${{ runner.os }}-pip${{ matrix.python-version }}
+        #- name: Persistent Github pip cache
+        #  uses: actions/cache@v2
+        #  with:
+        #    path: ${{ steps.pip-cache.outputs.dir }}
+        #    key: ${{ runner.os }}-pip${{ matrix.python-version }}
     - name: Upgrade pip
       run: python -m pip install --upgrade pip
     - name: Install nox

--- a/.github/workflows/on-push.yml
+++ b/.github/workflows/on-push.yml
@@ -77,18 +77,6 @@ jobs:
         key: ${{ runner.os }}-pip${{ matrix.python-version }}
     - name: Upgrade pip
       run: python -m pip install --upgrade pip
-    # macOS: Install wheel to allow wheel-building for nox and regex
-    # Build regex wheel into pip cache, so future pipx installs don't compile.
-    # This avoids future compile errors in nox tests because of missing
-    #   header files.
-    # Inside a pytest session, tests/conftest.py removes all existing paths from
-    #   the system PATH, which can cause C compilation to fail.  So here we
-    #   pre-build some packages' wheels outside of pytest.
-    - name: Install wheel and build wheel for regex to pip cache
-      if: ${{ runner.os == 'macOS' }}
-      run: |
-        python -m pip install wheel
-        python -m pip install regex
     - name: Install nox
       run: pip install nox
     - name: Execute Tests

--- a/.github/workflows/on-push.yml
+++ b/.github/workflows/on-push.yml
@@ -70,11 +70,11 @@ jobs:
       id: pip-cache
       run: |
         echo "::set-output name=dir::$(pip cache dir)"
-        # - name: Persistent Github pip cache
-        #   uses: actions/cache@v2
-        #   with:
-        #     path: ${{ steps.pip-cache.outputs.dir }}
-        #     key: ${{ runner.os }}-pip${{ matrix.python-version }}
+    - name: Persistent Github pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: ${{ runner.os }}-pip${{ matrix.python-version }}
     - name: Upgrade pip
       run: python -m pip install --upgrade pip
     - name: Install nox

--- a/noxfile.py
+++ b/noxfile.py
@@ -35,7 +35,7 @@ def tests(session):
         #   They will end up in the pip cache.
         session.install("wheel")
         for prebuild_package in macos_prebuild_packages:
-            session.run("pip", "wheel", prebuild_package)
+            session.run("pip", "wheel", prebuild_package, silent=True)
     session.install("-e", ".", "pytest", "pytest-cov")
     tests = session.posargs or ["tests"]
     session.run(

--- a/noxfile.py
+++ b/noxfile.py
@@ -23,8 +23,8 @@ lint_dependencies = [
     "check-manifest",
     "packaging>=20.0",
 ]
-# Packages that need to be compiled and need an intact system PATH on macOS
-macos_prebuild_packages = ["regex", "argon2-cffi"]
+# Packages that need an intact system PATH to compile on macOS
+macos_prebuild_packages = ["argon2-cffi"]
 
 
 @nox.session(python=python)
@@ -32,9 +32,10 @@ def tests(session):
     if sys.platform == "darwin":
         # For macOS, we need /usr/bin in PATH to compile some packages,
         #   but pytest setup clears PATH.  So we pre-build some wheels first.
+        #   They will end up in the pip cache.
         session.install("wheel")
         for prebuild_package in macos_prebuild_packages:
-            session.install(prebuild_package)
+            session.run("pip", "wheel", prebuild_package)
     session.install("-e", ".", "pytest", "pytest-cov")
     tests = session.posargs or ["tests"]
     session.run(

--- a/noxfile.py
+++ b/noxfile.py
@@ -29,7 +29,7 @@ lint_dependencies = [
 def tests(session):
     if sys.platform == "darwin":
         # For macOS, we need /usr/bin in PATH to compile some packages,
-        #   but tests clear PATH.  So we pre-build wheels here first.
+        #   but pytest setup clears PATH.  So we pre-build some wheels first.
         session.install("wheel")
         session.install("regex")
         session.install("argon2-cffi")

--- a/noxfile.py
+++ b/noxfile.py
@@ -27,6 +27,12 @@ lint_dependencies = [
 
 @nox.session(python=python)
 def tests(session):
+    if sys.platform == "darwin":
+        # For macOS, we need /usr/bin in PATH to compile some packages,
+        #   but tests clear PATH.  So we pre-build wheels here first.
+        session.install("wheel")
+        session.install("regex")
+        session.install("argon2-cffi")
     session.install("-e", ".", "pytest", "pytest-cov")
     tests = session.posargs or ["tests"]
     session.run(

--- a/noxfile.py
+++ b/noxfile.py
@@ -23,6 +23,8 @@ lint_dependencies = [
     "check-manifest",
     "packaging>=20.0",
 ]
+# Packages that need to be compiled and need an intact system PATH on macOS
+macos_prebuild_packages = ["regex", "argon2-cffi"]
 
 
 @nox.session(python=python)
@@ -31,8 +33,8 @@ def tests(session):
         # For macOS, we need /usr/bin in PATH to compile some packages,
         #   but pytest setup clears PATH.  So we pre-build some wheels first.
         session.install("wheel")
-        session.install("regex")
-        session.install("argon2-cffi")
+        for prebuild_package in macos_prebuild_packages:
+            session.install(prebuild_package)
     session.install("-e", ".", "pytest", "pytest-cov")
     tests = session.posargs or ["tests"]
     session.run(

--- a/noxfile.py
+++ b/noxfile.py
@@ -24,7 +24,7 @@ lint_dependencies = [
     "packaging>=20.0",
 ]
 # Packages that need an intact system PATH to compile on macOS
-macos_prebuild_packages = ["argon2-cffi"]
+macos_prebuild_packages = ["argon2-cffi", "regex"]
 
 
 @nox.session(python=python)

--- a/noxfile.py
+++ b/noxfile.py
@@ -27,15 +27,22 @@ lint_dependencies = [
 macos_prebuild_packages = ["argon2-cffi", "regex"]
 
 
+def prebuild_wheels(session, package_list):
+    wheel_dir = Path(session.virtualenv.location) / "prebuild_wheels"
+    wheel_dir.mkdir()
+    session.install("wheel")
+    for prebuild_package in package_list:
+        session.run(
+            "pip", "wheel", f"--wheel-dir={wheel_dir}", prebuild_package, silent=True,
+        )
+
+
 @nox.session(python=python)
 def tests(session):
     if sys.platform == "darwin":
-        # For macOS, we need /usr/bin in PATH to compile some packages,
-        #   but pytest setup clears PATH.  So we pre-build some wheels first.
-        #   They will end up in the pip cache.
-        session.install("wheel")
-        for prebuild_package in macos_prebuild_packages:
-            session.run("pip", "wheel", prebuild_package, silent=True)
+        # For macOS, need /usr/bin in PATH to compile some packages, but pytest
+        #   setup clears PATH.  So pre-build some wheels to the pip cache.
+        prebuild_wheels(session, macos_prebuild_packages)
     session.install("-e", ".", "pytest", "pytest-cov")
     tests = session.posargs or ["tests"]
     session.run(

